website/project/model.py

```diff
@@ -282,7 +282,11 @@ def get_content(self, auth):
                                 or (auth and not auth.user.is_anonymous() and self.user._id != auth.user._id)):
             return None
 
-        return self.content
+        content = self.content
+        if has_anonymous_link(self.node, auth):
+            content = re.sub('\[@[^\]]*\]\([^\) ]*\)', '@A User', content)
```

Comment:
I'm not necessarily opposed to having this in the model, but I think you should be able to handle this in the serializer. You'd make a field type like AnonymizedRegexField (or you might just make a field composition, depending).